x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/cli/ErrorsTestCase.java

```diff
@@ -37,15 +37,30 @@ public void testSelectFromMissingIndex() throws IOException {
     }
 
     @Override
-    public void testSelectFromIndexWithoutTypes() throws Exception {
+    public void testSelectColumnFromMissingIndex() throws Exception {
+        assertFoundOneProblem(command("SELECT abc FROM test"));
+        assertEquals("line 1:17: Unknown index [test]" + END, readLine());
+    }
+
+    @Override
+    public void testSelectFromEmptyIndex() throws Exception {
         // Create an index without any types
         Request request = new Request("PUT", "/test");
         request.setJsonEntity("{}");
         client().performRequest(request);
 
         assertFoundOneProblem(command("SELECT * FROM test"));
-        //assertEquals("line 1:15: [test] doesn't have any types so it is incompatible with sql" + END, readLine());
-        assertEquals("line 1:15: Unknown index [test]" + END, readLine());
+        assertEquals("line 1:8: Cannot determine columns for [*]" + END, readLine());
```

Comment:
I think this is incorrect - if we know the index is empty, we should return no columns since we can determine them and know there are none.

Reply:
Created https://github.com/elastic/elasticsearch/issues/53001